taskvine/src/manager/vine_file.c

```diff
@@ -103,11 +103,28 @@ should not be cached. That is, generating a cachename could lead to possible con
 
 
 */
-const char * make_mt_cached_name(const struct vine_file *f){
+struct vine_file * retrieve_url(const struct vine_file *f){
+	
+	char source_enc[PATH_MAX];
+	url_encode(path_basename(f->source), source_enc, PATH_MAX);
+	char * command = string_format("curl -sSL --stderr /dev/stdout -o \"%s\" \"%s\"", source_enc, f->source);
+	FILE * stream = popen(command, "r");
+	if(stream  == 0) return 0; // failed to execute curl
+	int exit_status = pclose(stream);
+	if(exit_status == 6) return 0; // curl could not resolve host;
+	if(exit_status == 0){
+		return vine_file_create(source_enc, f->remote_name, 0, 0, VINE_FILE, f->flags, 0);
+	}
+	else {
+		return 0;
+	}
+	return 0;
+}
+
+char * vine_task_to_json(const struct vine_file *f){
 	char * buffer;
 	char * file_buffer;
 	struct vine_file * file;
-	unsigned char digest[MD5_DIGEST_LENGTH];
 	buffer = string_format("{\ncmd = \"%s\"\n", f->source);
 	if(f->mini_task->input_files){
 		buffer = string_combine(buffer, "inputs = ");
```

Comment:
One thing that we have to be careful of here is that lists need to appear in the same order every time, so that we get consistent hashes of mini tasks.  At least drop a comment here so we don't forget to do it.